### ISSUE TITLE
Different time when date is typed in the input field

### ISSUE THREAD

**Issue by belmirp** [gpbl/react-day-picker#875]
`<DayPickerInput placeholder="DD/MM/YYYY" format="DD/MM/YYYY" onDayChange={day => console.log(day)} />`

If you click on a day from the dropdown, you get value:

> Thu Feb 28 2019 12:00:00 GMT+0100 (Central European Standard Time)

If you type the same date, you get this value:

> Thu Feb 28 2019 00:00:00 GMT+0100 (Central European Standard Time)

In some scenarios, e.g. when comparing selected value to `new Date()`, this could give wrong result.

**Comment by gpbl** [comment]
Yeah I also would expect the time being set to `12:00:00`.

**Comment by Way-U-Need** [comment]
you can use inputProps and set readOnly property so that user have to select from dropdown and can not type manually to avoid this conflict

**Comment by cherezovvv** [comment]
until a patch is released with fix #881, you can do the following
```
<DayPickerInput 
    placeholder="DD/MM/YYYY" 
    format="DD/MM/YYYY" 
    onDayChange={day => {
        if (day instanceof Date) {
            day.setHours(12);
        }
        console.log(day);
    } 
/>
```
